keps/672-execution-policy/README.md

```diff
@@ -0,0 +1,401 @@
+                - name: model-initializer
+                  persistentVolumeClaim:
+                    claimName: model-initializer
+    - name: trainer-node
+      template:
+        spec:
+          parallelism: 3
+          completions: 3
+          template:
+            spec:
+              containers:
+                - name: trainer
+                  image: docker.io/kubeflow/deepspeed-trainer
+                  resources:
+                    limits:
+                      nvidia.com/gpu: 5
+                  volumeMounts:
+                    - mountPath: /workspace/dataset
+                      name: dataset-initializer
+                    - mountPath: /workspace/pre-trained-model
+                      name: model-initializer
+              volumes:
+                - name: dataset-initializer
+                  persistentVolumeClaim:
+                    claimName: dataset-initializer
+                - name: model-initializer
+                  persistentVolumeClaim:
+                    claimName: model-initializer
+```
+
+#### Story 3
+
+As an HPC user, I want to run a set of simulations with MPI that complete successfully
+(or some percentage) before using the data for a next step analysis.
+
+As an HPC user I want to scale up my simulation job only after a small set have completed
+successfully. For that case, I could imagine essentially the same replicatedJob done twice,
+just with a larger size. If the outcome of the first isn't success you wouldn't launch
+the larger bulk of work.
+
+### Risks and Mitigations
+
+This API will not allow to describe DAGs to avoid workflow manager features in JobSet.
+The goal is to only focus on Job sequence to cover model training/HPC use-cases.
+
+## Design Details
+
+### API Details
+
+```golang
+type JobSetSpec struct {
+	ExecutionPolicy *ExecutionPolicy `json:"executionPolicy,omitempty"`
+}
+
+type ExecutionPolicyOption string
+
+const (
+	// This is the default settings.
+	// AnyOrder means that Jobs will be started in any order.
+	AnyOrder ExecutionPolicyOption = "AnyOrder"
+
+	// InOrder starts the ReplicatedJobs in order that they are listed. Jobs within a ReplicatedJob
+	// will still start in any order.
+	InOrder ExecutionPolicyOption = "InOrder"
+)
+
+type ExecutionPolicy struct {
+	// Order in which Jobs will be created.
+	ExecutionPolicyOrder ExecutionPolicyOption `json:"executionPolicyOrder"`
+
+	// After all ReplicatedJobs reach this status, the JobSet will create the next ReplicatedJobs.
+	ExecutionPolicyRule []ExecutionPolicyRule `json:"rules"`
+}
+
+// ExecutionPolicyRule represents the execution policy rule for Job sequence.
+type ExecutionPolicyRule struct {
+
+	// Names of the replicated Jobs that applied the status.
+	TargetReplicatedJobs []string `json:"targetReplicatedJobs"`
+
+	// Status the target ReplicatedJobs must reach before subsequent ReplicatedJobs begin executing.
+	ReplicatedJobsStatus ReplicatedJobsStatusOption `json:"replicatedJobsStatus"`
```

Comment:
I think, making this as a list makes sense to me. Any thoughts @danielvegamyhre @kannon92 @vsoch ?

Reply:
I guess, if we only support Ready + Succeeded it doesn't make sense to put list here, since Succeeded always comes after Ready status.
The list might be useful if in the future, we also support the Failed condition.
In that case, user can say that I want my ReplicatedJobs to be Succeeded or Failed before processing to the next ReplicatedJobs.

Reply:
>In that case, user can say that I want my ReplicatedJobs to be Succeeded or Failed before processing to the next ReplicatedJobs.

Rather than representing this as a list of ["Succeeded OR Failed"], we could represent this as a "Completed" status (succeeded or failed) and just keep the field as a single specific status to wait for.

Reply:
Completed status looks good to me.